### PR TITLE
Enable persist in specified schema

### LINKED ISSUE #945: allow the schema to be specified in --persist
Allow schema to be specified when saving a df using `--persist`

Context :
image

### PR DESCRIPTION
**Pull request by neelasha23**
## Issue number

Closes #945 

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--947.org.readthedocs.build/en/947/

### CODE REVIEW

**Comment by neelasha23** [comment]
There is some issue with the `Trino` docker. I can see that the container is being initialized but then a connection error keeps coming up. The same config is working fine on my local. So I'm not sure what exactly the issue is. 
I have added the `--persist` using schema steps in the trino user guide.

The rest of the code is ready for review. I will look into the issue if the Trino integration tests are mandatory for this PR. Please let me know @edublancas

**Comment by edublancas** [comment]
we're relying on pandas for this feature, right? It's ok to skip the integration tests for this one, let's just ensure it works for sqlite, and duckdb. it should be fine in other databases as this is just calling a pandas function.

once this is in, we can ask the user who opened the issue to verify that this is working

**Comment by edublancas** [comment]
please clean up the code to remove the integration tests and add sqlite/duckdb tests

**Comment by neelasha23** [comment]
> please clean up the code to remove the integration tests and add sqlite/duckdb tests

sqlite doesnt support schema. i have added integration tests for duckdb, postgrtes, snowflake. Will clean up the code.

File: src/sql/connection/connection.py [hunk 1]
```diff
@@ -1052,7 +1053,7 @@ def connection_sqlalchemy(self):
             "This feature is only available for SQLAlchemy connections"
         )
 
-    def to_table(self, table_name, data_frame, if_exists, index):
+    def to_table(self, table_name, data_frame, if_exists, index, schema):
```
Comment (edublancas) [inline src/sql/connection/connection.py:1056]: let's set schema=None for consistency

File: src/sql/connection/connection.py [hunk 2]
```diff
@@ -492,7 +492,7 @@ def _get_database_information(self):
         pass
 
     @abc.abstractmethod
-    def to_table(self, table_name, data_frame, if_exists, index):
+    def to_table(self, table_name, data_frame, if_exists, index, schema):
```
Comment (edublancas) [inline src/sql/connection/connection.py:495]: let's set schema=None for consistency

File: doc/integrations/pandas.md [hunk 3]
```diff
@@ -104,6 +104,8 @@ will create a table name in the database from the named DataFrame.   Or use `--a
 %sql SELECT * FROM df;
 ```
 
+A schema can also be specified when using the `--persist` argument. [Click here](../integrations/trinodb.ipynb) for more details.
```
[the remark this answers is not in the document]
Comment (neelasha23) [reply doc/integrations/pandas.md:107]: this file uses `sqlite` and schema is not suported in `sqlite`. should I add a section at the end with duckdb?

[the remark this answers is not in the document]
Comment (edublancas) [reply doc/integrations/pandas.md:107]: yes, please add it with duckdb

File: src/sql/connection/connection.py (hunk 2 above)
Comment (neelasha23) [reply src/sql/connection/connection.py]: [fixed](https://github.com/ploomber/jupysql/pull/947/files#diff-62e3d6289e3242b271e6df3a7f201ab003f8ce73a5daf1a824cc2b4d892513d1R495-R496)

File: doc/integrations/pandas.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (neelasha23) [reply doc/integrations/pandas.md]: fixed

**Review by edublancas — APPROVED** [review]
please remember to request a review, otherwise I don't know if this is ready or not (I assumed it is since all observations were addressed)

**Pull request merged**